CMSgov/cmcs-eregulations
# EREGCSC-1606 -- Two Column Search

Resolves [EREGCSC-1606](https://jiraent.cms.gov/browse/EREGCSC-1606)

**Description**
Two column keyword search.  

Left column is Regulations search results and show exactly mirror the current functionality of prod.  

Right column is Resources search results and should exactly match the same keyword search (without filters) on the Resources page.

Current behavior with search suggestions (quoted phrases, synonyms) should apply to both columns.

Empty state should match existing Resources empty state, but without the first link that links to either Regulations Search or Resources search depending on where you are.  There should only be eCFR and Federal Register links.

Style should match the Figma comp below.

[Figma comp here](https://www.figma.com/proto/iZFh6UhvGVEHw1D4IAt3Bw/eRegs-Combined-Search?page-id=3295%3A11315&node-id=3295%3A11490&viewport=-7943%2C4747%2C0.28&scaling=scale-down&starting-point-node-id=3295%3A11342&show-proto-sidebar=1)

**This pull request changes:**

- Refactors Resources page to pull common functionality out of the Resources page and creates new, atomic Vue components that are shared between Resources and Search page. Including:
   - `SearchInput.vue`
   - `ResourceResults.vue`
- Creates new Vue components for Regulations search results, including `RegResultsItem.vue` and `RegResults.vue`
- Extends existing shared components (like `SearchEmptyState.vue`) to be used better in both Resources and Search pages.
- creates very basic `getRegSearchResults` api method for requesting regulation search results from the front end.
- moves `getFormattedPartsList` method from the Resources page to `api.js` as it is used in both Search and Resources.
   - This is a big, ugly method that pulls a ton of ToC data in order to collect all information needed for the Part dropdown on the Resources page, as well as the information needed to properly construct links to the Related Regulations.  This is due for a refactor.
- Refactors `getTagContent` method from Python on the back end to JavaScript on the front end.

**Steps to manually verify this change:**
1. Visit experimental deployment here: https://c2g1vgggkk.execute-api.us-east-1.amazonaws.com/dev622
2. In the top right corner of the page, enter a search term into the search box and press enter
3. Ensure that the Search page loads properly, has two columns, and the results look as expected
   - compare Reg Search results to another experimental branch's Reg Search results
   - compare Resources results to the results for the same keyword in the Resources view
4. Poke around Search page, click synonyms, clear searches, copy and paste URL of a search into another browser tab.  Make sure empty states match the Figma comp.
